Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -832,19 +876,19 @@ extension ProjectDescription.Settings {
             settingsDictionary["SWIFT_ACTIVE_COMPILATION_CONDITIONS"] = .array(["$(inherited)"] + swiftDefines)
         }
 
-        if !cFlags.isEmpty {
+        if !cFlags.isEmpty, target.type.supportsCSettings {
```

Comment:
Can't we have a supportsCustomSettings and skip the whole `try settings.forEach` above, instead of skipping pieces here one by one?

Reply:
skipping `forEach` might result in the processed target potentially missing some settings, e.g. `MODULEMAP_FILE` and others which are still required I think

Reply:
The forEach should be only for settings defined in the Package.swift, which shouldn't be supported for systemLibrary I think, the `MODULEMAP_FILE` and other settings are handled after that, isn't it?

Reply:
oh, right. So you mean something like this?

```swift
if target.type.supportsCustomSettings {
  try settings.forEach { setting in ... }
}
```

Reply:
yep!

Reply:
done